integration_tests/README.md

```diff
@@ -303,3 +303,94 @@ The marks you care about are all in marks.py
 
 For the most part you can ignore this file. It provides the underlying Spark session to operations that need it, but most tests should interact with
 it through `asserts.py`.
+
+## Guidelines for Testing
+
+When support for a new operator is added to `rapids-plugin-4-spark`, it is recommended that
+the following conditions be covered in its corresponding integration tests:
+
+### 1. Cover all supported data types
+Ensure that tests cover all data types supported by the added operation. An exhaustive list of data types supported in 
+Spark is available [here](https://spark.apache.org/docs/latest/sql-ref-datatypes.html). These include:
+   * Numeric Types 
+     * `ByteType` 
+     * `ShortType` 
+     * `IntegerType`
+     * `LongType`
+     * `FloatType`
+     * `DoubleType`
+     * `DecimalType`
+   * Strings 
+     * `StringType` 
+     * `VarcharType`
+   * Binary (`BinaryType`)  
+   * Booleans (`BooleanType`)
+   * Chrono Types 
+     * `TimestampType` 
+     * `DateType`
+     * `Interval`
+   * Complex Types 
+     * `ArrayType`
+     * `StructType`
+     * `MapType`
+
+`data_gen.py` provides `DataGen` classes that help generate test data in integration tests.
+
+### 2. Nested Data types
+Complex data types (`ArrayType`, `StructType`, `MapType`) warrant extensive testing for various combinations of nesting.
+E.g.
+   * `Array<primitive_type>`
+   * `Array<Array<primitive_type>>`
+   * `Array<Struct<primitive_type>>`
+   * `Struct<Array<primitive_type>>`
+   * `Array<Struct<Array<primitive_type>>>`
+   * `Struct<Array<Struct<primitive_type>>>`
+
+The `ArrayGen` and `StructGen` classes in `data_gen.py` can be configured to support arbitrary nesting.
+
+### 3. Scalar values
+Operators and expressions that support scalar operands need to be tested with scalar inputs, of all 
+supported types from 1 and 2, above. 
+For instance, `SUM()` supports numeric columns (e.g. `SUM(a + b)`), or scalars (e.g. `SUM(20)`).
+Similarly, `COUNT()` supports the following:
+   * Columns: E.g. `COUNT(a)` to count non-null rows for column `a`
+   * Scalars: E.g. `COUNT(1)` to count all rows (including nulls)
+   * `*`: E.g. `COUNT(*)`, functionally equivalent to `COUNT(1)`
+It is advised that tests be added for all applicable scalar types, for an operator.
+
```

Comment:
Can we add a comment about static folding in Spark?  Something like 


For many operations if all of the inputs are scalar/literal values Apache Spark will evaluate that expression and replace it with a new literal value in the logical planning/optimization stage. So if you type in `5 + 10` Spark will replace it with a literal `15` before the plugin sees it.  This is great for performance, but typically does not really add more test coverage.

Reply:
Excellent point! I'll mention this.

I'm not familiar with the Spark vernacular. Shall I refer to this optimization as "Static Folding" in the documentation? I'm struggling to find references to link to. (Hive calls this "Constant Folding".)

Reply:
```
  /**
   * Returns true when an expression is a candidate for static evaluation before the query is
   * executed. A typical use case: [[org.apache.spark.sql.catalyst.optimizer.ConstantFolding]]
   *
   * The following conditions are used to determine suitability for constant folding:
   *  - A [[Coalesce]] is foldable if all of its children are foldable
   *  - A [[BinaryExpression]] is foldable if its both left and right child are foldable
   *  - A [[Not]], [[IsNull]], or [[IsNotNull]] is foldable if its child is foldable
   *  - A [[Literal]] is foldable
   *  - A [[Cast]] or [[UnaryMinus]] is foldable if its child is foldable
   */
  def foldable: Boolean = false
```

And the name of the optimization rule is `ConstantFolding` so my guess is that constant folding is probably the better term to use.

Reply:
Agreed. I've added this description. I've tried to keep it short so as to keep the checklist manageable.